native/cocos/base/StringUtil.cpp

```diff
@@ -86,12 +86,19 @@ int StringUtil::printf(char *buf, const char *last, const char *fmt, ...) {
 }
 
 ccstd::string StringUtil::format(const char *fmt, ...) {
-    char sz[4096];
     va_list args;
     va_start(args, fmt);
-    vprintf(sz, sz + sizeof(sz) - 1, fmt, args);
+    // Caculate the string length
+    int length = vsnprintf(nullptr, 0, fmt, args);
+    if (length <= 0) {
```

Comment:
Does passing 0 will return the string length? From https://cplusplus.com/reference/cstdio/vsnprintf/, I don't find the information.

Reply:
It is better to add a unit test for this function.

Reply:
Oh, just saw the unit test.

Reply:
I found a FreeBSD version of vsnprintf.c which is opensource by Apple , you could refer to 

https://github.com/apple-oss-distributions/Libc/blob/7861c72b1692b65f79c03f21a8a1a8e51e14c843/stdio/FreeBSD/vsnprintf.c#L73

Reply:
It works on android too. Need to check whether it works on openharmony later.

Reply:
Works on OpenHarmony, too.